projects/ng-event-plugins/src/events/longtap.event.ts

```diff
@@ -0,0 +1,142 @@
+import type {EventManager} from '@angular/platform-browser';
+
+import type {EventManagerPlugin} from '../types/event-manager-plugin';
+import {LongtapEvent} from '../types/longtap.event';
+import {isIos} from '../utils/is-ios';
+
+const TAP_DELAY = 700;
+const SAFE_NAVIGATOR = typeof navigator === 'undefined' ? null : navigator;
+const MOVE_THRESHOLD = 15;
+
+export class LongtapEventPlugin implements EventManagerPlugin {
+    private readonly isIOS = !!SAFE_NAVIGATOR && isIos(SAFE_NAVIGATOR);
+
+    public readonly manager!: EventManager;
+
+    public addEventListener(
+        element: HTMLElement,
+        _event: string,
+        handler: (event: Event) => void,
+    ): Function {
+        const removeLongtapEventPolyfill = this.isIOS
+            ? this.listenTouchEvents(element)
+            : this.listenContextmenuEvent(element);
+
+        element.addEventListener('longtap', handler);
+
+        return () => {
+            removeLongtapEventPolyfill();
+
+            element.removeEventListener('longtap', handler);
+        };
+    }
+
+    public supports(event: string): boolean {
+        return /^longtap$/i.test(event);
+    }
+
+    private listenContextmenuEvent(element: HTMLElement): Function {
+        return this.manager.addEventListener(
+            element,
+            'contextmenu.prevent.stop',
```

Comment:
Oh, I see you are stopping native `contextmenu` as well. Are we sure we want that?

Reply:
I think so, because there is no way to catch this event in the upstream component. When it listening for a longtap event, the plugin will recreate all listeners.

Reply:
Ok, makes sense